src/quantuminspire/qiskit/circuit_parser.py

```diff
@@ -17,170 +17,410 @@
+            instruction: The Qiskit instruction to translate to cQASM.
 
-        Returns:
-            cQASM code string.
         """
-        qubit_indices = tuple(circuit['qubits'])
-        return 'Z q[{0}]\n'.format(*qubit_indices)
+        qubit_indices = tuple(instruction.qubits)
+        stream.write('Z q[{0}]\n'.format(*qubit_indices))
 
-    def _u(self, circuit: Dict[str, Any]) -> str:
-        """ Translates the U element to U3.
+    @staticmethod
+    def _c_z(stream: StringIO, instruction: QobjItem, binary_control: str) -> None:
+        """ Translates the binary-controlled Z element.
 
         Args:
-            circuit: The Qiskit circuit with U element.
+            stream: The string-io stream to where the resulting cQASM is written.
+            instruction: The Qiskit instruction to translate to cQASM.
+            binary_control: The multi-bits control string. The gate is executed when all specified classical bits are 1.
 
-        Returns:
-            cQASM code string.
         """
-        return self._u3(circuit)
+        qubit_indices = tuple(instruction.qubits)
+        stream.write('C-Z {0}q[{1}]\n'.format(binary_control, *qubit_indices))
 
-    def _u0(self, circuit: Dict[str, Any]) -> None:
-        """ Translates the U0 element. Not used!
+    @staticmethod
+    def _r(stream: StringIO, instruction: QobjItem, axis: str) -> None:
+        """ Translates the Rotation element for an axis (x,y,z).
 
         Args:
-            circuit: The Qiskit circuit with U0 element.
+            stream: The string-io stream to where the resulting cQASM is written.
+            instruction: The Qiskit instruction to translate to cQASM.
+            axis: The axis for which the Rotation operator is parsed ('x', 'y' or 'z').
 
-        Returns:
-            None.
         """
-        return None
+        angle_q0 = instruction.params[0]
+        qubit_indices = tuple(instruction.qubits)
+        stream.write('R{0} q[{1}], {2:.6f}\n'.format(axis, *qubit_indices, angle_q0))
 
-    def _u1(self, circuit: Dict[str, Any]) -> str:
+    @staticmethod
+    def _c_r(stream: StringIO, instruction: QobjItem, axis: str, binary_control: str) -> None:
+        """ Translates the binary-controlled Rotation element for an axis (x,y,z).
+
+        Args:
+            stream: The string-io stream to where the resulting cQASM is written.
+            instruction: The Qiskit instruction to translate to cQASM.
+            axis: The axis for which the Rotation operator is parsed ('x', 'y' or 'z').
+            binary_control: The multi-bits control string. The gate is executed when all specified classical bits are 1.
+
+        """
+        angle_q0 = instruction.params[0]
+        qubit_indices = tuple(instruction.qubits)
+        stream.write('C-R{0} {1}q[{2}], {3:.6f}\n'.format(axis, binary_control, *qubit_indices, angle_q0))
+
+    def _rx(self, stream: StringIO, instruction: QobjItem) -> None:
```

Comment:
Ok. I made them all static now, solving [DEM-542], but I doubt if this is better OO. There is in my opinion no need to make the members static because an instance of CircuitToString is available when the parse method is called.